src/SMU-Agilent_B29xx/main.py

```diff
@@ -93,11 +106,44 @@ def get_GUIparameter(self, parameter={}):
         # if self.average > 100:
         #     self.average = 100
 
-        self.device = parameter['Device']
-        self.channel = str(parameter['Channel'])[-1]
+        self.device = parameter["Device"]
+        self.channel = str(parameter["Channel"])[-1]
+
+        # check in GUI to select pulse option
+        self.pulse = parameter["CheckPulse"]
+
+        # for use on the B2902B, defined as the delay for measurement start after pulse release;
+        # variable not yet implemented
+        # self.pulse_meas_time_in_s = parameter["PulseMeasStart_in_s"]
+
+        # for use on the B2902B, defined as the pulse width time
+        self.ton = float(round(float(parameter["PulseOnTime"]), 6))
```

Comment:
float-round-float? What exactly is the intention? Floating point numbers don't have fixed decimal lenght (because they are binary).

I.e. the exact value of `float(round(float((0.0012123)), 6))`  is `0.0012119999999999999697186670033488553599454462528228759765625`, because 0.001212 does not have an exact representation as float. In most of the cases it will probably work, because str(...) of a float will be rounded as well, so the result looks like it only has 6 decimals. But I'm not sure if that's true in every scenario.

Fun fact: Calculate `x = 0.1 + 0.1 + 0.1 - 0.3` in Python ^^

Reply:
This is part of the code taken over from the existing Keithley 26xx SMU driver when the implementation of the pulse capabilities began and was used as a starting point. It is non-sense and was just overlooked as the pick-up of the pulse on-time out of the GUI wasn't toutched again after it was first integrated. Just change it to the way self.toff is implemented and modify it in the Keithley driver as well.

Reply:
I have removed the outer float(), but kept the round(..., 6). Not sure why it was added to the Keithley 26xx driver, but I guess the idea was to prevent floating numbers with many digits and to limit the smallest step to 1 µs. 

Added round(..., 6) to toff as well.